Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1

```diff
@@ -1625,6 +1625,335 @@ function Test-DiscoveryService{
     }
 }
 
+function Migrate-AADSettings {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string] $installConfigPath,
+        [Parameter(Mandatory=$true)]
+        [string] $azureConfigPath
+    )
+    $configSection = "identity"
+    $nodesToSearch = @("tenants", "replyUrls", "claimsIssuerTenant", "allowedTenants", "registeredApplications")
+    $childNodeGetAttribute = "name"
+    $childNodeSetAttribute = "value"
+
+    Write-DosMessage -Level "Verbose" -Message "Get the AAD Setting contents in install.config"
+    
+    # Quick check to see if any AAD settings are present in install.config (short circuit)
+    $existingAADSettings = Check-XMLChildNode -installConfigPath $installConfigPath -configSection $configSection -nodeToSearch $nodesToSearch[0] -childNodeGetAttribute $childNodeGetAttribute
+    if($false -eq $existingAADSettings)
+    {
+      Write-DosMessage -Level "Verbose" -Message "No AAD Settings found in install.config"
+      return
+    }
+
+    # Gather AAD Settings from install.config
+    $existingChildNodes = Get-XMLChildNodes -installConfigPath $installConfigPath -configSection $configSection -nodesToSearch $nodesToSearch -childNodeGetAttribute $childNodeGetAttribute
+    $azureSecretNameFromInstallConfig = Get-XMLChildNode -installConfigPath $installConfigPath -configSection $configSection -childNodeGetAttribute $childNodeGetAttribute -childNodeAttributeSetting "azureSecretName" 
+    $childNodeHeader = @{}
+    $childNodeHeader.Add("secretName","")
+    $existingChildNodes += $childNodeHeader
+    $existingChildNodes += $azureSecretNameFromInstallConfig
+    Write-DosMessage -Level "Verbose" -Message "Copied the AAD Settings from install.config"
+
+    # Remove blank child nodes and azureSecretName from default azuresettings.config manifest
+    Write-DosMessage -Level "Verbose" -Message "Started removing blank child variables and azureSecretName from azuresettings.config in nuget folder"
```

Comment:
This might not always be in the nuget folder if a migrationAzureConfigPath parameter is passed to the script right?
Would writing the azureConfigPath instead of 'nuget folder' be better here?

Reply:
Yah, only if it is run outside of the DosInstaller.